sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_message_backcompat.py

```diff
@@ -48,45 +72,49 @@ class MessageAlreadySettled(Exception):
 
 
 class LegacyMessage(object):  # pylint: disable=too-many-instance-attributes
-    def __init__(self, message, **kwargs):
+    def __init__(
+        self,
+        message: "AmqpAnnotatedMessage",
+        **kwargs: Any
+    ) -> None:
         self._message: "AmqpAnnotatedMessage" = message
-        self.state = MessageState.SendComplete
-        self.idle_time = 0
-        self.retries = 0
-        self._settler = kwargs.get("settler")
+        self.state: "MessageState" = MessageState.SendComplete
+        self.idle_time: int = 0
+        self.retries: int = 0
+        self._settler: "Settler" = kwargs.pop("settler")
```

Comment:
I don' think we can do a Pop here as it will break EventHUB

Reply:
since we dont settle

Reply:
```suggestion
try:
        self._settler: "Settler" = kwargs.pop("settler")
except KeyError:
      pass
```?

Reply:
in EH, does the message error when you call one of the properties that call this self._settler? or does it return None?

Reply:
errors when called

Reply:
sorry, I meant to say, does it error in EH with uamqp?

Reply:
aye aye captain